Data/Scripts/001_Technical/002_RubyUtilities.rb

```diff
@@ -162,6 +159,145 @@ def self.move(source, destination)
+  # new constructor accepts RGB values as well as a hex number or string value
+  #-----------------------------------------------------------------------------
+  def initialize(*args)
+  	pbPrintException("Wrong number of arguments! At least 1 is needed!") if args.length < 1
+  	if args.length == 1
+      if args[0].is_a?(Fixnum)
+        hex = args[0].dup.to_s(16)
+      elsif args[0].is_a?(String)
+        hex = args[0].dup
+        hex.gsub!("#", "") if hex.include?("#")
+      end
+      pbPrintException("Wrong type of argument given!") if !hex
+      r = hex[0...2].to_i(16)
+      g = hex[2...4].to_i(16)
+      b = hex[4...6].to_i(16)
+  	elsif args.length == 3
+      r, g, b = *args
+  	end
+  	return init_original(r, g, b) if r && g && b
+  	return init_original(*args)
+  end
+  #-----------------------------------------------------------------------------
+  # returns an RGB color value as a hex value
+  #-----------------------------------------------------------------------------
+  def to_hex
+  	r = sprintf("%02X", self.red)
+  	g = sprintf("%02X", self.green)
+  	b = sprintf("%02X", self.blue)
+  	return ("#" + r + g + b).upcase
+  end
+  #-----------------------------------------------------------------------------
+  # returns Hex color value as RGB
+  #-----------------------------------------------------------------------------
+  def to_rgb(hex)
+  	hex = hex.to_s(16) if hex.is_a?(Numeric)
+  	r = hex[0...2].to_i(16)
+  	g = hex[2...4].to_i(16)
+  	b = hex[4...6].to_i(16)
+  	return r, g, b
+  end
+  #-----------------------------------------------------------------------------
+  # returns decimal color
+  #-----------------------------------------------------------------------------
+  def to_dec
+  	return self.to_hex.delete("#").to_i(16)
+  end
+  #-----------------------------------------------------------------------------
+  # byte order in ARGB instead of RGBA.
+  #-----------------------------------------------------------------------------
+  def to_i
+    return self.alpha.to_i << 24 | self.blue.to_i << 16 | self.green.to_i << 8 | self.red.to_i
+  end
+  #-----------------------------------------------------------------------------
+  # return color from byte integer value
+  #-----------------------------------------------------------------------------
+  def self.from_i(color)
+    b =  color & 255
+    g = (color >> 8) & 255
+    r = (color >> 16) & 255
+    a = (color >> 24) & 255
+    return Color.new(r, g, b, a)
+  end
+  #-----------------------------------------------------------------------------
+  # parse color input to return color object
+  #-----------------------------------------------------------------------------
+  def self.parse(color)
+    if color.is_a?(Color) # color object
+      return color
+    elsif color.is_a?(String) # string
+      if color.include?("#") # hex color
+        return Color.new(color)
+      elsif color.include(",") # RGB color
```

Comment:
Couldn't support for a string containing three comma-separated values go in `def initialize` instead?

Reply:
It could probably go there as well, but not instead. To keep parsing and init capability the same.